backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/RemoveImageCommand.java

```diff
@@ -323,4 +337,32 @@ protected VDSReturnValue performDeleteImageVdsmOperation() {
                                 getDiskImage().isWipeAfterDelete(), getStorageDomain().getDiscardAfterDelete(),
                                 getParameters().isForceDelete())));
     }
+
+    @Override
+    public CommandCallback getCallback() {
+        return callbackProvider.get();
+    }
+
+    public void onSucceeded() {
+        performImageDbOperations();
+    }
+
+    public void onFailed() {
+        if (getParameters().getAsyncTaskErrorMessage() != null) {
+            List<Guid> snapshotIds = diskImageDao
+                    .getAllSnapshotsForImageGroup(getParameters().getDiskImage().getId()).stream()
+                    .map(snapshot -> snapshot.getImageId())
```

Comment:
Some minor stuff:
1. `snapshot -> snapshot.getImageId()` --> `DiskImage::getImageId` is more elegant :)
2. Maybe worth creating the local variable to shorten some code and eliminate duplication:
`Guid diskId = getParameters().getDiskImage().getId())`.
BTW: what is `getParameters().getImageId()` , is it the same?

Reply:
Ack
No, it's not the same. 
getImageId() returns the active volume ID
getDiskImage.getId() returns disk ID

Reply:
Ack, can resolve this one as well :)